### PR TITLE
feat: integrate blobstore in pool

### PR DESCRIPTION
**Pull request by mattsse**
another incremental change that adds the blobstore to the Pool type

needs a few followups, but best to keep these small

### CODE REVIEW

File: crates/transaction-pool/src/metrics.rs [hunk 1]
```diff
@@ -38,6 +38,16 @@ pub struct TxPoolMetrics {
     pub(crate) performed_state_updates: Counter,
 }
 
+/// Transaction pool blobstore metrics
+#[derive(Metrics)]
+#[metrics(scope = "transaction_pool")]
+pub struct BlobStoreMetrics {
+    /// Number of failed inserts into the blobstore
```
Comment (onbjerg) [inline crates/transaction-pool/src/metrics.rs:45]: Should we also have other metrics? Blob size, inserts, removes etc. (not only failures)

Comment (mattsse) [reply crates/transaction-pool/src/metrics.rs:45]: will definitely add

but would like to include these prs asap

File: crates/transaction-pool/src/lib.rs [hunk 2]
```diff
@@ -86,6 +87,13 @@
 //! that provides the `TransactionPool` interface.
 //!
 //!
+//! ## Blob Transactions
+//!
+//! Blob transaction can be quite large hence they are stored in a separate blobstore. The pool is
+//! responsible for inserting blob data for new transactions into the blobstore.
+//! See als [ValidTransaction](validate::ValidTransaction)
```
Comment (Rjected) [inline crates/transaction-pool/src/lib.rs:94]: typo nit: `see also`

**Review by Rjected — APPROVED** [review]
LGTM

**Pull request merged**